lms/djangoapps/teams/static/teams/js/teams_features.js

```diff
@@ -51,13 +51,24 @@
         });
     };
 
-    function removeBrowseTab(teams){
-        if (teams.count == 1){
+    function removeBrowseAndButtons(staff, teamsLocked){
+        if (teamsLocked && !staff){
+            // Remove browse
             $("#tab-1").remove();
             $("#tabpanel-browse").remove();
             $("#tab-0").addClass("is-active");
             $("#tabpanel-my-teams").removeClass("is-hidden");
-        }
+            // remove join and leave button
+            $(".join-team .action-primary").remove();
+            $(".page-content-secondary .leave-team").remove();
+
+        }else if(teams.count == 1 && !staff){
```

Comment:
This method does not receive anymore the var teams, but it's using the var on the method. The reason why I think it works is because there is a var "teams" defined in the code and is valid in the function scope when it gets executed. Better if we pass the "teams" parameter? another approach is to keep the method removeBrowseTab and calling it from inside removeBrowseAndButtons in the case when teams are not locked. The advantage of this approach is that we don't delete removeBrowseTab method, just add a new one, so we have only additive changes

Reply:
I have change this, please check it